mdoc-sbt/src/main/scala/mdoc/MdocPlugin.scala

```diff
@@ -54,23 +54,40 @@ object MdocPlugin extends AutoPlugin {
   }
   val mdocInternalVariables =
     settingKey[List[(String, String)]](
-      " variables that can be referenced from markdown with @VERSION@."
+      "Additional site variables that are added by mdoc plugins. Not intended for public use."
     )
   import autoImport._
 
+  lazy val validateSettings = Def.task[Unit] {
+    val in = mdocIn.value.toPath
+    val base = baseDirectory.value.toPath
+    if (in == base) {
```

Comment:
@vlovgr this guard is technically too strict, it's OK for `baseDirectory` to match `mdocIn` as long as `mdocOut` is customize to generate sources elsewhere. However, I feel more comfortable if `mdocIn` doesn't contain any generated stuff like `target/`, even if it can be worked around with exclude filters. What do you think?

Reply:
In terms of avoiding the surprise of growing input sources, this works great. 👍 Personally, I quite like that everything relating to docs is in a single directory. This includes `src` (e.g. for using the library API), markdown files, and `target`. The benefit of not moving `mdocOut` is that we can clean it up with `docs/clean`.

I ended up with setting `mdocIn` to `src/main/mdoc`, which I think is better than relying on exclude filters, and it works with this check, and `docs/clean`. It is also the only change needed to be able to use `docs` as the directory. I believe this is what tut also does (uses `src/main/tut`). Is there a case for having `src/main/mdoc` as the default input directory?

Reply:
Moving `mdocIn` to `src/main/mdoc` is a good solution 👍 I personally prefer having markdown in the topelevel `docs/` directory since that makes it easier to discover for people who clone the repo for the first time. Docusaurus, read the docs, mkdocs, sphinx and several other tools have `docs/` as the default input directory.

Reply:
Yeah, I agree it's preferable. I guess the difference is that, since it's an sbt project, there are other things going on in that same directory (minimum generating `target`). For my curiosity, what's preventing mdoc from copying `target` to `target` with `docs/` as input source?

Having `src/main/mdoc` as default input plays nice with sbt projects. There's also the case that the markdown files needs pre-processing and aren't 'final output' that e.g. Docusaurus should read (which is why we set the `customDocsPath`).

Just some food for thought. I'm happy with manually setting `mdocIn`. :)

Reply:
My past experience with sbt plugins is that I always end up regretting going against sbt idioms, which means `src/main/mdoc` is probably a better default... will think about it